packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -723,18 +723,40 @@ class FileCollection extends DocumentCollection {
 
   async downloadArchive(fileIds, notSecureFilename = 'files') {
     const filename = slugify(notSecureFilename)
-    const href = await this.getArchiveLinkByIds(fileIds, filename)
+    const href = await this.getArchiveLinkByIds({
+      ids: fileIds,
+      name: filename
+    })
     const fullpath = this.stackClient.fullpath(href)
     this.forceFileDownload(fullpath, filename + '.zip')
   }
 
-  async getArchiveLinkByIds(ids, name = 'files') {
+  /**
+   * Attributes used for get archive link by ids
+   *
+   * @typedef {object} ArchivePages
+   * @property {string} id - Id of the file
+   * @property {number} page - The page number
+   */
+
+  /**
+   * Get the archive link for a list of files
+   * The generated archive is temporary and is not persisted
+   *
+   * @param {object} params - Parameters
+   * @param {string[]} params.ids - List of file ids
+   * @param {string} [params.name] - Name of the archive (default: 'files')
+   * @param {ArchivePages[]} [params.pages] - Array of objects, with `id` the file identifier, and `page` the page number (1 is the first page)
+   * @returns {Promise<string>} - The archive link
+   */
+  async getArchiveLinkByIds({ ids, name = 'files', pages }) {
```

Comment:
I would have avoided a breaking change. For example, we can keep the `getArchiveLinkByIds` as it was, marks it as deprecated in the documentation, and add a new function `createArchiveLinkByIds`.

Reply:
I like that, and the name will be more accurate too 👍